gdal/ogr/swq_op_general.cpp

```diff
@@ -1182,7 +1182,7 @@ swq_field_type SWQGeneralChecker( swq_expr_node *poNode,
             eRetType = SWQ_STRING;
             eArgType = SWQ_STRING;
         }
-        else if( poNode->papoSubExpr[0]->field_type == SWQ_FLOAT )
+        else if( poNode->papoSubExpr[0]->field_type == SWQ_FLOAT || poNode->papoSubExpr[1]->field_type == SWQ_FLOAT )
```

Comment:
I believe the below test for SWQ_INTEGER64 at l 1190 (and the one at l 1214 too) would also benefit from the same logic, since currently cast(1 as integer)+ 1234567890123 isn't promote to Integer64

Reply:
Yes. Indeed